apps/workflows/src/cron/checker.ts

```diff
@@ -0,0 +1,216 @@
+import { CloudTasksClient } from "@google-cloud/tasks";
+import type { google } from "@google-cloud/tasks/build/protos/protos";
+import { z } from "zod";
+
+import { and, db, eq, gte, lte, notInArray } from "@openstatus/db";
+import {
+  maintenance,
+  maintenancesToMonitors,
+  monitor,
+  type monitorStatusSchema,
+  monitorStatusTable,
+  selectMonitorSchema,
+  selectMonitorStatusSchema,
+} from "@openstatus/db/src/schema";
+
+import type { monitorPeriodicitySchema } from "@openstatus/db/src/schema/constants";
+import type { httpPayloadSchema, tpcPayloadSchema } from "@openstatus/utils";
+import { env } from "../env";
+
+// We can't secure cron endpoint thus we should make sure they are called by the generated url
+// FIXME: we can with CRON_SECRET!
+export const isAuthorizedDomain = (url: string) => {
```

Comment:
I'm not sure it's still useful ?

Reply:
if we are going to use google cloud scheduler we can set some auth header 
![CleanShot 2024-11-18 at 14 08 18@2x](https://github.com/user-attachments/assets/019a2906-2b95-4c6d-9fa9-580ff15c0825)